crates/sui-framework/sources/bcs.move

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::bcs {
+    use std::vector as v;
+
+    /// For when bytes length is less than required for deserialization.
+    const EOutOfRange: u64 = 0;
+
+    /// Read address from the bcs-serialized bytes.
+    public fun peel_address(bcs: &mut vector<u8>): address {
+        assert!(v::length(bcs) >= 20, EOutOfRange);
```

Comment:
Might want an `const SUI_ADDRESS_LENGTH: u64 = 20` for readability + to prevent copypasta errors if someone tries to use this elsewhere.